storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -602,7 +607,15 @@
                     (log-message "Activating spout " component-id ":" (keys task-datas))
                     (fast-list-iter [^ISpout spout spouts] (.activate spout)))
                
-                  (fast-list-iter [^ISpout spout spouts] (.nextTuple spout)))
+                  (if (and
+                        ((:storm-conf executor-data) TOPOLOGY-BACKPRESSURE-ENABLE)
+                        @(:throttle-on (:worker executor-data))
+                        suspend-time
+                        (not= suspend-time 0))
+                    (do 
```

Comment:
Can we move this conditional up to be part of the overflow-buffer/max-spout-pending check?  It feels like it fits better there, and that we don't want to output anything when back-pressure is on, instead of just slowing down how quickly we output.

Also instead of logging something, can we look at having a metric instead that shows how may times we called paused for a given reason.  Logging is not going to really give us the picture we want and is going to be difficult to follow.